wapitiCore/attack/mod_basic_auth_bf.py

```diff
@@ -0,0 +1,187 @@
+from itertools import chain
+from os.path import join as path_join
+import json
+import requests
+from bs4 import BeautifulSoup
+from requests.exceptions import ReadTimeout, RequestException
+from wapitiCore.net.web import Request
+from wapitiCore.attack.attack import Attack, Mutator
+from wapitiCore.language.vulnerability import Vulnerability, Anomaly, _
+
+
+class mod_basic_auth_bf(Attack):
+    name = "basic_auth_bf"
+    payloads_username = []
+    payloads_password=[]
+    is_success = False
+    PAYLOADS_FILE = "passwords.txt"
+    PAYLOADS_FILE_USER = "users.txt"
+    PAYLOADS_SUCCESS = "successMessage.txt"
+    PAYLOADS_FAIL = "incorrectMessage.txt"
+    MSG_VULN = _("Brute force attack success")
+    current_request_url = None
+    current_request_method = None
+    password_parameter = []
+    username_parameter_field = []
+    submit_var_name = None
+    submit_var_value = None
+
+    def set_timeout(self, timeout):
+        self.time_to_sleep = str(1 + int(timeout))
+
+    def is_false_positive(self, request):
+        try:
+            response = self.crawler.send(request)
+        except RequestException:
+            pass
+        else:
+            if self._find_pattern_in_response(response.content):
+                return True
+        return False
+
+    def check_html_content(self):
+        has_password_field = False
+        content = requests.get(self.current_request_url).content
+        inputs = BeautifulSoup(content, features="lxml").findAll('input')
+        for _input in inputs:
+            if _input.attrs['type'] == 'submit':
+                self.submit_var_name = _input.attrs['name']
+                self.submit_var_value = _input.attrs['name']
+            if _input.attrs['type'] == 'password' and self.current_request_method == 'POST':
+                self.password_parameter.append(_input.attrs['name'])
+                # TODO OUSS more controls for this field
+                self.username_parameter_field.append(inputs[inputs.index(_input) - 1].attrs['name'])
+                has_password_field = True
+        return has_password_field
+
+    def check_success_auth (self, content_response:str) :
+        self.is_success = False
```

Comment:
This variable is unnecessary, remove it from the code.

Reply:
fixed